datasource/http/data.go

```diff
@@ -125,7 +125,7 @@ func (d *Datasource) Execute() (cty.Value, error) {
 
 	defer resp.Body.Close()
 
-	if resp.StatusCode != 200 {
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
nit: Is it helpful to use the net/http constants? 
```suggestion
	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http. StatusMultipleChoices {
```

Reply:
Yeah that's fair. Feel free to merge when ready.